src/test/fuzz/fuzz.cpp

```diff
@@ -12,13 +12,16 @@
 
 #include <cstdint>
 #include <exception>
+#include <functional>
 #include <memory>
 #include <string>
 #include <unistd.h>
 #include <vector>
 
 const std::function<void(const std::string&)> G_TEST_LOG_FUN{};
 
+const std::function<std::vector<const char*>()> G_TEST_COMMAND_LINE_ARGUMENTS{};
+
```

Comment:
are these lines supposed to be here? the commit message mentions only unit tests so i thought it was odd to see changes to `fuzz.cpp` and then i noticed these lines are removed in the next commit

Reply:
Yes because every commit has to be self-sufficient - transform the tree from one valid state to another valid state (where it compiles, tests pass, there are no bugs, etc). If `G_TEST_COMMAND_LINE_ARGUMENTS` is not defined in `src/test/fuzz/fuzz.cpp` in the first commit, then that commit (without the subsequent ones) will not compile due to undefined reference. The next commit just changes the value of the variable.

Reply:
huh, interesting. might be worth squashing the commits and changing the commit message to indicate you are changing unit and fuzz?

then again, if no one else has mentioned it, probably not a big deal

Reply:
I would keep it as is - the first commit introduces command line parsing from unit tests. As a necessity, it adds stubs in `src/bench/bench.cpp`, `src/qt/test/test_main.cpp` and `src/test/fuzz/fuzz.cpp` because without those stubs the compilation would fail. The subsequent commits are not necessary for the first one - it can be merged alone, should reviewers reject e.g. the command line parsing from fuzz tests.